overlay_test.go

```diff
@@ -138,6 +138,92 @@ relations:
 	c.Assert("\n"+string(overlayYaml), gc.Equals, expOverlay)
 }
 
+func (*bundleDataOverlaySuite) TestExtractAndMergeWithMixedOverlayBits(c *gc.C) {
+	// In this example, mysql defines an offer whereas wordpress does not.
+	//
+	// When the visitor code examines the application map, it should report
+	// back that the filtered "mysql" application key should be retained
+	// but the "wordpress" application key should NOT be retained. The
+	// applications map should be retained because at least one of its keys
+	// has to be retained. However, the "wordpress" entry must be removed.
+	// If not, it would be encoded as an empty object which the overlay
+	// merge code would mis-interpret as a request to delete the "wordpress"
+	// application from the base bundle!
+	data := `
+bundle: kubernetes
+applications:
+  mysql:
+    charm: cs:mysql
+    scale: 1
+    offers:
+      my-offer:
+        endpoints:
+        - apache-website
+        - website-cache
+        acl:
+          admin: admin
+          foo: consume
+  wordpress:
+    charm: cs:wordpress
+    scale: 2
+relations:
+- - wordpress:db
+  - mysql:mysql
+`
+
+	expBase := `
+bundle: kubernetes
+applications:
+  mysql:
+    charm: cs:mysql
+    series: kubernetes
+    num_units: 1
+  wordpress:
+    charm: cs:wordpress
+    series: kubernetes
+    num_units: 2
+relations:
+- - wordpress:db
+  - mysql:mysql
+`
+
+	expOverlay := `
+applications:
+  mysql:
+    offers:
+      my-offer:
+        endpoints:
+        - apache-website
+        - website-cache
+        acl:
+          admin: admin
+          foo: consume
+`
+
+	bd, err := charm.ReadBundleData(strings.NewReader(data))
+	c.Assert(err, gc.IsNil)
+
+	base, overlay, err := charm.ExtractBaseAndOverlayParts(bd)
+	c.Assert(err, jc.ErrorIsNil)
+
+	baseYaml, err := yaml.Marshal(base)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert("\n"+string(baseYaml), gc.Equals, expBase)
```

Comment:
I'd rather express this as:
```
expected := `
blah
blah
`[1:]
```
rather than "\n" + got.
It is a common pattern we use elsewhere, so I think people will understand and expect it.

Reply:
Sure thing.